pkg/apis/core/types.go

```diff
@@ -3523,6 +3523,11 @@ type PodStatus struct {
 	// Status for any ephemeral containers that have run in this pod.
 	// +optional
 	EphemeralContainerStatuses []ContainerStatus
+
+	// TotalResourcesRequested is the effective resource requirements for this pod, taking into consideration init containers, sidecars, containers and pod overhead.
```

Comment:
lets not mentione sidecars before they are implemented

```suggestion
	// TotalResourcesRequested is the effective resource requirements for this pod, taking into consideration init and regular containers, and the pod overhead.
```

Reply:
please explain how this behaves if request is not set